src/components/introduce/FullPageCarousel.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import useEmblaCarousel from 'embla-carousel-react';
+import * as React from 'react';
+import {
+  carouselContainer,
+  carouselSlide,
+  carouselWrapper,
+  progressBar,
+  progressBarContainer,
+} from './introduce.css';
+
+const FullPageCarousel = Object.assign(BaseCarousel, { Slide: CarouselSlide });
+
+export default FullPageCarousel;
+
+export function BaseCarousel({ children }: React.PropsWithChildren) {
+  const [emblaRef, emblaApi] = useEmblaCarousel();
+
+  const slideCount = React.Children.count(children);
+  const [progressRate, setProgressRate] = React.useState((1 / slideCount) * 100);
+
+  React.useEffect(() => {
+    const updateProgressRate = () => {
+      const currentIndex = (emblaApi?.selectedScrollSnap() || 0) + 1;
+
+      setProgressRate((currentIndex / slideCount) * 100);
+    };
+
+    emblaApi?.on('select', updateProgressRate);
+
+    return () => {
+      emblaApi?.off('select', updateProgressRate);
+    };
+  }, [emblaApi, slideCount]);
+
+  return (
+    <div>
+      <div className={progressBarContainer}>
+        <div className={progressBar} style={{ width: `${progressRate}%` }}></div>
+      </div>
```

Comment:
style 객체를 따로 만들어주신 이유가 동적스타일링 때문에 그런것일까요?!!

Reply:
네네 progressRate에 따라 너비를 조정하기 위해 인라인을 썼습니다 더 좋은 방법이 있을까요?

Reply:
```suggestion
<div className={progressBarContainer}>
          <div className={`${progressBar} ${getProgressStyle(progressRate)}`} ></div>;
          </div>
```
```suggestion
  export const getProgressStyle = (progressRate: number) =>
  style({
    width: `${progressRate}%`,
  });
```
이런 방법을 생각했었는데,  이 방법대로 하면 `progressRate`가 바뀔 때마다 클래스명도 같이  바뀌니 CSS가 계속 누적되면서 성능 저하를 일으킬수 도 있을 것 같네요...

한영님이 말씀하신 것처럼 인라인을 사용하는 것이 가장 적합할 것 같네요..

Reply:
사실 저도 처음에 이 방식으로 시도했었는데요 ㅎㅎ 에러가 발생하더라구요! 함수 형태로 export 할 수 없다는 에러였어요

Reply:
그런 또 문제점이 있었군요.. 하나 더 알아갑니다 🥹